### DISCUSSION TITLE
collection.products filter by productType not working

### DISCUSSION THREAD

**Discussion by nmoncur** [Shopify/storefront-api-feedback#120] [Help]
Can reproduce on the 2022-10 and 2022-07 API versions in the storefront GraphQL playground [here](https://shopify.dev/graphiql/storefront-graphiql), it seems that filtering by productType isn't working. Or I'm a noob, always possible... I am fairly new to GraphQL, let me know if there is another way to filter a collection by product type that I might be missing...

Here's the query:

```
{
  collection(handle: "latest-stuff") {
    id
    handle
    title
    products(first: 100, filters: {productType: "Dress"}) {
      nodes {
        title
        productType
      }
    }
  }
}
```

And the results display all varying product types:

```
{
  "data": {
    "collection": {
      "id": "gid://shopify/Collection/389242179",
      "handle": "latest-stuff",
      "title": "Latest Stuff",
      "products": {
        "nodes": [
          {
            "title": "DKNYPURE PONCHO TOP",
            "productType": "Blouse"
          },
          {
            "title": "DKNYPURE TEE WITH UNDERLAYER",
            "productType": "Top"
          },
          {
            "title": "DOUBLE RACER BACK DRESS",
            "productType": "Dress"
          },
          {
            "title": "DKNYPURE DRAPE NECK DRESS",
            "productType": "Dress"
          },
          {
            "title": "DKNYPURE SHORT SLEEVE TEE",
            "productType": "Top"
          },
          {
            "title": "DKNYPURE RACERBACK TANK DRESS",
            "productType": "Dress"
          },
          {
            "title": "COLOR BLOCK SHEATH DRESS",
            "productType": "Dress"
          },
          {
            "title": "COLOR BLOCK CREWNECK DRESS",
            "productType": "Dress"
          },
          {
            "title": "DKNYPURE V-NECK PULLOVER",
            "productType": "Sweater"
          }
        ],
     ...
```

**Comment by xxdd13** [comment]
I've got a similar issue, filter by `productVendor` is not working too.
I've tried on 2022-01 to 2022-10.
```gql
query {
  collection(id: "gid://shopify/Collection/389242179") {
    handle
    products(first: 10, filters: {productVendor: "ABC"}) {
      edges {
        node {
          id
          vendor
        }
      }
    }
  }
}

```
<img width="1122" alt="Screenshot 2022-07-18 at 12 53 40 am" src="https://user-images.githubusercontent.com/26791482/179404077-885af210-f823-44d6-9bc2-0b69d17c2360.png">

**Reply by dev2919** [reply]
+1 
I have a similar issue going on. Trying to filter with variantOption but it doesnt seem to work

![Screenshot 2022-07-25 at 12 48 57 PM](https://user-images.githubusercontent.com/48598352/180720422-6889d577-697b-4538-ae8a-0b32405df828.png)

**Reply by parthshiroya6** [reply]
can you please sent filters using tags example Like, filters: {tags:"XYZ"}  because i can't filter any product using tags

**Comment by code-by-gijs** [comment]
Did you enable the product type in the "Collection and search filters" section in the Shopify admin?

**Reply by dev2919** [reply]
@code-by-gijs cant access this section as the store is on shopify lite plan :/ 
ps. some variant options do work as intended but recently added product variant dont get filtered at all.

**Reply by mtrofimiuk** [reply]
That was it. Steps to solve. 
Open Store Admin -> Sales Channels -> Online Store -> Navigation
Add "Search & Discovery App" by Shopify
Now you can edit "Collection and search filters" in the app settings.
